src/pages/MyPage/MyPageMain.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import Nav from '../../Components/Nav/Nav';
+import Footer from '../../Components/Footer/Footer';
+import MyPageCogIconControl from './Components/MyPageCogIconControl/MyPageCogIconControl';
+import { TEST_TOKEN, API_MY_PAGE_MAIN } from '../../config';
+import './MyPageMain.scss';
+
+class MyPageMain extends Component {
+  constructor() {
+    super();
+    this.state = {
+      myPageInfo: [],
+      titleSection: '',
+    };
+  }
+  componentDidMount = () => {
+    fetch(API_MY_PAGE_MAIN, {
+      method: 'GET',
+      headers: {
+        Authorization: TEST_TOKEN,
+      },
+    })
+      .then(response => response.json())
+      .then(result => {
+        this.setState({ myPageInfo: result });
+      });
+  };
+
+  handleClickEval = () => {
+    this.props.history.push({
+      pathname: '/mypagefooterdetail',
+      state: { titleSection: '평가한 작품들' },
+    });
+  };
+
+  handleClickWish = () => {
+    this.props.history.push({
+      pathname: '/mypagefooterdetail',
+      state: { titleSection: '보고싶어요' },
+    });
+  };
+
+  handleClickWatching = () => {
+    this.props.history.push({
+      pathname: '/mypagefooterdetail',
+      state: { titleSection: '보는중' },
+    });
+  };
```

Comment:
하나의 함수에서 관리할 수 있습니다
클릭하는 div에 id 설정하시고 e.target.id를 전달해서 해당 배열의 인덱스값을 전달하는 방법입니다.
그리고 titleSection을 state에서 관리할 필요가 없어보입니다.

```suggestion
  handleClickEval = (e) => {
       const myPageState = ["평가한 작품들", "보고싶어요", "보는중"];
   
     this.props.history.push({
      pathname: '/mypagefooterdetail',
      state: { titleSection: myPageState[Number(e.target.id)]  },
    });
  };
  
  ...
  
  return (
  
  
  ...
    <div className="myPageEval" id="0" onClick={this.handleClickEval}>
  )
```

Reply:
넵